### PR TITLE
feat(pending): add already pending error

### LINKED ISSUE #287: Revert when new value corresponds to the pending value
From https://github.com/cantinasec/review-morpho-blue-1/issues/78

### PR DESCRIPTION
**Pull request by Rubilmax**
- Fixes #287 
- Fixes https://github.com/cantinasec/review-morpho-blue-1/issues/78

### CODE REVIEW

**Review by Jean-Grimal — APPROVED** [review]

File: src/MetaMorpho.sol [hunk 1]
```diff
@@ -202,6 +202,10 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         if (newTimelock > timelock) {
             _setTimelock(newTimelock);
         } else {
+            if (pendingTimelock.submittedAt != 0 && newTimelock == pendingTimelock.value) {
```
Comment (MerlinEgalite) [inline src/MetaMorpho.sol:205]: Is the first check necessary?

How can we have `pendingTimelock.value != 0` but `pendingTimelock.submittedAt == 0`

Comment (Rubilmax) [reply src/MetaMorpho.sol:205]: I believe we are not looking to study `pendingTimelock.submittedAt == 0 && pendingTimelock.value != 0` but instead `pendingTimelock.submittedAt == 0 && pendingTimelock.value == newTimelock`

We know invariant `pendingTimelock.submittedAt == 0 && pendingTimelock.value == 0`

For some inputs, `newTimelock` may be `0` which would satisfy `pendingTimelock.submittedAt == 0 && pendingTimelock.value == newTimelock == 0`

I'll think for each of the 4 timelocked values:
- timelock: `0 < MIN_TIMELOCK <= newTimelock <= MAX_TIMELOCK` so impossible
- fee: `newFee` may equal `0`
- guardian: `newGuardian` may equal `address(0)`
- cap: `newCap` may equal `0`

Comment (MerlinEgalite) [reply src/MetaMorpho.sol]: Indeed for the `fee`, `guardian` and `cap`

Comment (MerlinEgalite) [reply src/MetaMorpho.sol]: well in fact I'm not sure, if `newFee` is 0, then it's not pending no? Same for the cap, right?

**Review by MerlinEgalite — APPROVED** [review]

**Comment by MerlinEgalite** [comment]
@StErMi this is the same PR as for the URD, can you have a look please?

**Comment by MerlinEgalite** [comment]
@Rubilmax can you update this PR please?

**Review by MerlinEgalite — APPROVED** [review]

**Review by Jean-Grimal — APPROVED** [review]
Clean

**Review by StErMi — APPROVED** [review]

**Review by Rubilmax — COMMENTED** [review]
LGTM

**Pull request merged**